packages/react/src/components/Tile/Tile.js

```diff
@@ -549,7 +579,8 @@ export class ExpandableTile extends Component {
       },
       () => {
         this.setMaxHeight();
-        this.props.handleClick(evt);
+        // TODO: Remove handleClick prop when handleClick is deprecated
+        this.props.handleClick?.(evt) || this.props.onClick(evt);
```

Comment:
Should the same optional chaining exist for `onClick` so we only call it if they pass it in? Would prevent the need for the no-op defaultProp functions.